lib/ansible/cli/galaxy.py

```diff
@@ -1206,6 +1231,17 @@ def execute_remove(self):
 
         return 0
 
+    def execute_remove_collection(self):
+        collections = context.CLIARGS['args']
+        search_paths = context.CLIARGS['collections_path']
+        ignore_errors = context.CLIARGS['ignore_errors']
+        requirements_file = context.CLIARGS['requirements']
+        no_deps = context.CLIARGS['no_deps']
+
+        requirements = self._require_one_of_collections_requirements(collections, requirements_file)['collections']
```

Comment:
What's the behaviour when you have a requirements file of and a different version of that collection is installed

```yaml
collections:
- name: community.general
  version: 1.0.0
```

Looking through the code it seems like it will skip this collection due to [this check](https://github.com/ansible/ansible/pull/71800/files#diff-2de3c8cc3e7fae814d02b3058f0c8f9eR621). Is that the desired behaviour or should we be removing all the collections specified regardless of the version in a requirements file?

Reply:
That is intentional. Using that requirements file to remove a collection would be equivalent to `ansible-galaxy collection remove community.general:1.0.0` or `ansible-galaxy collection remove community.general:==1.0.0`; if a version other than 1.0.0 is installed, it won't be removed. Does that seem like surprising behavior?